text/0036-exactness-criterion.md

```diff
@@ -0,0 +1,149 @@
+
+> Additionally, synonym matching and plural/singular matching are considered exact. Thus, a word is considered an exact match if its synonym matches a query exactly.
+
+Adding some exceptions by default:
+
+[...] single-word matches on multi-word attributes aren’t considered exact matches by default [...]
+[...] Multiword synonyms are not counted as exact matches by default [...]
+
+It is possible to specify that multi-words synonyms are considered an exact match using `alternativesAsExact` setting. Algolia also permit to disable some attributes to be used for exact criterion with `disableExactOnAttributes`setting.
+
+### IV. Explanation
+
+#### Current MeiliSearch Behavior (0.20)
+
+For each word of the query we match documents:
+
+- without typo
+- without prefix
+- without ngrams
+- without multi-words synonyms
+- without word split
+
+> If the query contains only one word, only attribute containing this one word value can match.
+
+##### Pros & Cons
+
+This behavior of the Exact criterion gives a better rank to documents that match the words of the original query, ranking ngrams, complex synonyms, and word split after.
+
+In the specific case of a query containing only 1 word, only 1-word attributes matching will boost documents that have an attribute that exactly contains the query word.
+
+This is useful when an ID is searched but in others cases it could impact the relevancy of search results.
+
+Moreover, when users search with a multi-words query, attributes that exactly contains the query are not boosted.
+
+Because MeiliSearch is a search as you type engine, attributes that exactly begin by the original query should also be boosted.
+
+#### Milli Behavior (0.21)
+
+We will divide Exactness into three layers of ranking.
+
+##### Layer 1: Try to match exactly the query in an attribute
+
+Any document that has an attribute that contains exactly the query:
+
+- without typo
+- without prefix
+- without ngrams
+- without multi-words synonyms
+- without word split
+- with words in the right order
+
+has a rank of `0` (perfect match)
+
+##### Layer 2: Try to match exactly the query at the start of an attribute
+
+Any document that has an attribute that contains the query at the n firsts positions:
+
+- without typo
+- without prefix
+- without ngrams
+- without multi-words synonyms
+- without word split
+- with words in the right order
+
+> OR
+
+Any document that has an attribute that contains exactly the query:
+
+- without typo
+- without ngrams
+- without multi-words synonyms
+- without word split
+- with words in the right order
+
+has a rank of `1` (the user didn't finish to type the search but the final query could possibly be an exact match)
+
+##### Layer 3: Try to match exactly the word of the query anywhere in the document
+
+Any other document has a rank of `2 + 1` for each word that is not an exact match.
+
+A word is considered as an exact match when it matches:
+
+- without typo
+- without prefix
```

Comment:
What does `without prefix` mean?

Reply:
No word can match as a prefix. If my query contains `many` and a document contains `manythefish`, It's not counted as an exact match.

Reply:
Thanks @LegendreM!